DeviceBridgeE2E/src/device-bridge.spec.ts

```diff
@@ -311,3 +310,43 @@ test.serial('Test auth', async t => {
         });
     }
 });
+
+test.serial('Test restart', async t => {
+    // Create subscription to Azure Function
+    const callbackUrl = `${t.context.ctx.callbackUrl}&deviceId=${t.context.device.id}`;
+    await t.context.ctx.deviceBridgAPI.createConnectionStatusSubscription(t, t.context.device.id, callbackUrl)
+    await sleep(3000);
+    // Ensure get works
+    var response = await t.context.ctx.deviceBridgAPI.getConnectionStatusSubscription(t, t.context.device.id);
+    t.is(callbackUrl, response.body.callbackUrl)
+
+    // Ensure connection created event when a sub created
+    await t.context.ctx.deviceBridgAPI.createCMDSubscription(t, t.context.device.id, callbackUrl);
+    await sleep(3000);
+    // Ensure connection restarted when container restarts
+    await got.post<{ [name: string]: string }>(
+        t.context.ctx.restartApiUrl,
+        {
+            json: {
+                request: {
+                },
+            },
+            responseType: 'json',
+            headers: {Authorization:"Bearer " + t.context.ctx.restartBearerToken},
+        }
+    );
+
+    await sleep(1000);
+    var invocationValue = await t.context.ctx.deviceBridgAPI.getEcho(t, t.context.device.id);
```

Comment:
This tests seems to be very time-sensitive. If I understand correctly, it's gonna fail if you don't receive the `Disabled` event within a second. One way to reduce the time-dependence would be to use the `deviceReceivedAt` field of the event.
Create the sub and record the `deviceReceivedAt` of the connection, then restart and check again after some time if you got a new connection event with a different `deviceReceivedAt` (also, could you put the second check in a loop that checks every, let's say, 5sec for a maximum of 1min? Just so we don't strictly depend on the 30 second mark).